src/test/ui/issues/issue-39848.stderr

```diff
@@ -1,17 +1,22 @@
-error: expected `{`, found `foo`
+error: unexpected macro fragment: foo
   --> $DIR/issue-39848.rs:3:21
    |
 LL |         if $tgt.has_$field() {}
-   |         --          ^^^^^^--
-   |         |           |
-   |         |           expected `{`
-   |         |           help: try placing this code inside a block: `{ $field() }`
-   |         this `if` expression has a condition, but no block
+   |                     ^^^^^^ unexpected macro fragment
 ...
 LL |     get_opt!(bar, foo);
    |     ------------------ in this macro invocation
    |
    = note: this error originates in the macro `get_opt` (in Nightly builds, run with -Z macro-backtrace for more info)
+help: the nightly only `std::concat_idents` macro can be used to synthesize identifiers
+   |
+LL |         if $tgt.concat_idents!(has_, $field)() {}
```

Comment:
Limiting this suggestion to the method call position is not only arbitrary, but also incorrect - `concat_idents!` cannot be used in that position.

Maybe it makes sense to provide this suggestion in some positions, but I wouldn't personally bother, `concat_idents!` is generally pretty useless.